ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/AbstractRecord.java

```diff
@@ -473,9 +473,21 @@ public static AbstractRecord create (int type)
         }
 	    catch (final Throwable ex)
 	    {
-	        ex.printStackTrace();
-	        
-	        return null;
+			try {
+				Class rt = RecordType.typeToClass(type);
+				String className = (rt == null) ? "null" : rt.getName();
+				int mb = 1024 * 1024;
+				Runtime runtime = Runtime.getRuntime();
+				long used = (runtime.totalMemory() - runtime.freeMemory()) / mb;
+				long free = runtime.freeMemory() / mb;
+
+				tsLogger.i18NLogger.warn_coordinator_AbstractRecord_create_failure(
+						type, className, used, free, ex);
+			} catch (final Throwable t) {
+				ex.printStackTrace();
```

Comment:
This code was already present in the original code and I have not changed it.

Reply:
Yes, and with enhancing our code practice it will be nice to improve it.

Reply:
I agree with Ondra, I think this (new catch final Throwable t) handling should use the logger but I am about to add another comment you might disagree with Ondra

Reply:
Why not raise a separate issue for it since we have lots of code that prints stacktraces so picking on this PR is an ad hoc extra requirement that is orthogonal to the purpose of the PR.